src/app/core/components/archive-type-change-dialog/archive-type-change-dialog.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Observable } from 'rxjs';
+import { ArchiveVO, ArchiveType } from '@models/archive-vo';
+import { Component, Inject, OnInit } from '@angular/core';
+import { DialogRef, DIALOG_DATA } from '@root/app/dialog/dialog.service';
+import { ApiService } from '@shared/services/api/api.service';
+import { AnyCnameRecord } from 'dns';
+
+@Component({
+  selector: 'pr-archive-type-change-dialog',
+  templateUrl: './archive-type-change-dialog.component.html',
+  styleUrls: ['./archive-type-change-dialog.component.scss'],
+})
+export class ArchiveTypeChangeDialogComponent implements OnInit {
+  archive: ArchiveVO;
+  archiveType: ArchiveType;
+  archiveClose: Observable<void>;
+  public updating = false;
+
+  constructor(
+    private dialogRef: DialogRef,
+    private api: ApiService,
+    @Inject(DIALOG_DATA) public data: any
+  ) {
+    this.archive = this.data.archive;
+    this.archiveType = this.data.archiveType;
+    this.archiveClose = this.data.archiveClose;
+  }
+
+  ngOnInit(): void {}
+
+  public onDoneClick(): void {
+    this.archiveClose.subscribe();
+    this.dialogRef.close();
+  }
+
+  public async onConfirmClick() {
+    this.archive.type = this.archiveType;
+    this.updating = true;
+    try {
+      await this.api.archive.update(this.archive);
+    } catch {
+      // fail silently
```

Comment:
I think ideally we'd tell the user that an error occurred, though I defer to @meisekimiu on how we usually communicate that.

Reply:
I think ideally we would at least like to call the `MessageService` to show an error in the UI (though no other action is really necessary).